src/rust/engine/graph/src/entry.rs

```diff
@@ -743,6 +761,13 @@ impl<N: Node> Entry<N> {
     }
   }
 
+  pub fn is_running(&self) -> bool {
+    match *self.state.lock() {
+      EntryState::Running { .. } => true,
+      EntryState::Completed { .. } | EntryState::NotStarted { .. } => false,
```

Comment:
Maybe just `_`. You shouldn't be forced to edit this arm when you add a new non-Running state, or even just use `matches!(...)` to leave off the arm altogether.

Reply:
For a small number of states, I've found that it is helpful to force thinking about their relevance to any particular match. It's unlikely but if we were to introduce a `RunningUnderLock` or something, we'd want to consider that here.